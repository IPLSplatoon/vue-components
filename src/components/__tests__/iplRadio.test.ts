import IplRadio from '../iplRadio.vue';
import { mount } from '@vue/test-utils';

describe('IplRadio', () => {
    it('matches snapshot', () => {
        const wrapper = mount(IplRadio, {
            props: {
                modelValue: 'opt1',
                label: 'Radio!',
                name: 'test-radio',
                options: [
                    { name: 'Option One', value: 'opt1' },
                    { name: 'Option Two', value: 'opt2', disabled: false },
                    { name: 'Option Three', value: 'opt3', disabled: true }
                ]
            }
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('emits update event when new option is selected', async () => {
        const wrapper = mount(IplRadio, {
            props: {
                modelValue: 'opt1',
                label: 'Radio!',
                name: 'test-radio',
                options: [
                    { name: 'Option One', value: 'opt1' },
                    { name: 'Option Two', value: 'opt2', disabled: false },
                    { name: 'Option Three', value: 'opt3', disabled: true }
                ]
            }
        });

        const secondOption = wrapper.findAll('input').at(1);
        expect(secondOption).not.toBeUndefined();
        secondOption!.element.checked = true;
        await secondOption!.trigger('change');

        const emitted = wrapper.emitted('update:modelValue');
        expect(emitted).toBeTruthy();
        expect(emitted?.[0]).toEqual(['opt2']);
    });
});
