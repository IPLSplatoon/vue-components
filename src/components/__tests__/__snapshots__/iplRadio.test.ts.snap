// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IplRadio matches snapshot 1`] = `
"<div class="ipl-radio__wrapper"><label class="ipl-label">Radio!</label>
  <div class="ipl-radio__options">
    <div class="ipl-radio__option wrap-anywhere selected" data-test="option_opt1">Option One</div>
    <div class="ipl-radio__option wrap-anywhere" data-test="option_opt2">Option Two</div>
    <div class="ipl-radio__option wrap-anywhere disabled" data-test="option_opt3">Option Three</div>
  </div>
</div>"
`;
